compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -1045,7 +1045,10 @@ object desugar {
     val (nestedStats, topStats) = pdef.stats.partition(needsObject)
     if (nestedStats.isEmpty) pdef
     else {
-      val sourceName = ctx.source.file.name.takeWhile(_ != '.')
+      var fileName = ctx.source.file.name
+      val end = fileName.lastIndexOf('.') // drop extension
+      val start = fileName.lastIndexOf('.', end - 1) // drop prefix
+      val sourceName = fileName.substring(start, end)
```

Comment:
So what happens if there are dots in the name ? Do they get encoded to `$something` ?

Reply:
They did not but I changed that in the latest commit. `a.b` gets encoded to `a$u002Eb`.